src/GovUk.Education.ExploreEducationStatistics.Common/Services/DataTableStorageService.cs

```diff
@@ -0,0 +1,91 @@
+#nullable enable
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Data.Tables;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Services;
+
+public class DataTableStorageService(string tableStorageConnectionString) : IDataTableStorageService
+{
+    private readonly TableServiceClient _client = new(tableStorageConnectionString);
+
+    public async Task<TEntity?> GetEntityIfExists<TEntity>(
+        string tableName, 
+        string partitionKey, 
+        string rowKey,
+        IEnumerable<string>? select = null,
+        CancellationToken cancellationToken = default)
+        where TEntity : class, ITableEntity
+    {
+        var tableClient = await GetTableClient(tableName);
+
+        var createdEntity = await tableClient.GetEntityIfExistsAsync<TEntity>(
+            partitionKey: partitionKey,
+            rowKey: rowKey,
+            select: select,
+            cancellationToken: cancellationToken);
+
+        return createdEntity.HasValue
+            ? createdEntity.Value
+            : null;
+    }
+
+    public async Task CreateEntity(
+        string tableName, 
+        ITableEntity entity,
+        CancellationToken cancellationToken = default)
+    {
+        var tableClient = await GetTableClient(tableName);
+
+        await tableClient.AddEntityAsync(
+            entity: entity,
+            cancellationToken: cancellationToken);
+    }
+
+    public async Task UpdateEntity(
+        string tableName,
+        ITableEntity entity,
+        TableUpdateMode updateMode = TableUpdateMode.Replace,
+        CancellationToken cancellationToken = default)
+    {
+        var tableClient = await GetTableClient(tableName);
+
+        await tableClient.UpdateEntityAsync(
+            entity: entity,
+            ifMatch: entity.ETag,
+            mode: updateMode,
+            cancellationToken: cancellationToken);
+    }
+
+    public async Task DeleteEntity(
+        string tableName, 
+        string partitionKey,
+        string rowKey,
+        CancellationToken cancellationToken = default)
+    {
+        var tableClient = await GetTableClient(tableName);
+
+        await tableClient.DeleteEntityAsync(
+            partitionKey: partitionKey,
+            rowKey: rowKey,
+            ifMatch: ETag.All,
+            cancellationToken: cancellationToken);
+    }
+
+    /// <summary>
+    /// Gets a table by name, will create the table if it does not exist
+    /// </summary>
+    /// <param name="tableName">The name of the table to get.</param>
+    /// <returns>The table</returns>
+    private async Task<TableClient> GetTableClient(string tableName)
+    {
+        var tableClient = _client.GetTableClient(tableName);
+
+        await tableClient.CreateIfNotExistsAsync();
```

Comment:
I think this means that the change described in the infrastructure ticket EES-4927 to create the table is not required in advance.

That ticket also describes creating a new queue, but we do a similar thing for queues

We do a similar `CreateIfNotExists` for getting a handle on storage queues, see

https://github.com/dfe-analytical-services/explore-education-statistics/blob/afa5dd143813b0400360e34830e8bc0022bcd21b/src/GovUk.Education.ExploreEducationStatistics.Common/Services/QueueServiceClient.cs#L42-L47

I think that means you don't need this ticket at all?

Reply:
You're absolutely right! I think I'll repurpose that ticket to be a faffy-deploy ticket for the keyvault values we need